### DISCUSSION TITLE
Mimetype admin controls in realtionship field

### DISCUSSION THREAD

**Discussion by AkashReddy-playy** [payloadcms/payload#496] [Feature Requests & Ideas]
Hello,

Currently, mimeTypes are set in the collections.
Is there a possibility of fields admin controls to filter  Mimetypes of the upload relations? 
or any workarounds to be considered?
Any suggestions would be great.

**Comment by jmikrut** [comment]
Hey @AkashReddy-playy — we are actually working on a feature for the Relationship field right now to do exactly this—it will allow you to filter relationship field `options` within the admin UI as well as use the filter within the relationship validation itself to restrict relationships to only certain documents. This will work perfectly for you to restrict relationship fields by mime type.

Should be delivered shortly! I will post back here when it's released!

**Reply by AkashReddy-playy** [reply]
Hey @jmikrut,
That will be a great addition, much appreciate the support.

**Comment by DanRibbens** [comment]
Hi @AkashReddy-playy, we are close to having this ready. After https://github.com/payloadcms/payload/pull/506 is merged and release you will be able to configure `filterOptions` on a field to limit the options available.

Here is an example:
```
const pngOnlyFiles = {
      name: 'files',
      type: 'upload',
      relationTo: 'files',
      filterOptions: {
        // here we can limit the mimeType using a query.
        mimeType: { equals: 'image/png' },
      },
};
```

Look for it in the next release :)

**Reply by jmikrut** [reply]
Hey @AkashReddy-playy — this is now deployed in `payload@0.15.4`!

It's documented [here](https://payloadcms.com/docs/fields/relationship#filtering-relationship-options) but works exactly the same for `upload` fields.

Take a look!

**Reply by AkashReddy-playy** [reply]
Thanks, @DanRibbens @jmikrut  
Maybe we have missed out to test the implementation thoroughly. A consideration to filter multiple mimetypes would be a great help.
Great going  🙌.

**Reply by AkashReddy-playy** [reply]
Hi @DanRibbens, I believe the relationship component is broken after the update.
<img width="951" alt="Screenshot 2022-04-06 at 10 23 26 AM" src="https://user-images.githubusercontent.com/67790455/161898457-af52ad3f-dc98-47ff-9de3-bc6b62803af0.png">
.

**Reply by DanRibbens** [reply]
Ah! Thank you for reporting @AkashReddy-playy! I released patch `0.15.5` that addresses it.
